rclrs/src/node.rs

```diff
@@ -189,6 +190,17 @@ impl Node {
         Ok(client)
     }
 
+    /// Creates a [`GuardCondition`][1].
+    ///
+    /// [1]: create::GuardCondition
+    pub fn create_guard_condition(
+        &self,
+        context: &Context,
+        callback: Option<Box<dyn Fn() + Send + Sync>>,
+    ) -> Result<Arc<GuardCondition>, RclrsError> {
+        Ok(Arc::new(GuardCondition::new(context, callback)))
```

Comment:
This function should also add the guard condition to a node-internal list of guard conditions, which are then added to the wait set in `spin_once()`, like subscriptions etc.